### PR TITLE
Documenting Image Mechanics

### PR DESCRIPTION
**Pull request by cdunbar13**
### Submission Checklist

Please take the following actions before submitting this pull request.

* Fork your PR branch from the Toolkit "develop" branch (not main)
* Test all changes with pre-commit in a local branch [#](https://goo.gle/hpc-toolkit#development)
* Confirm that "make tests" passes all tests
* Add or modify unit tests to cover code changes
* Ensure that unit test coverage remains above 80%
* Update all applicable documentation
* Follow Cloud HPC Toolkit Contribution guidelines [#](https://goo.gle/hpc-toolkit-contributing)

### CODE REVIEW

File: docs/vm-images.md [hunk 1]
```diff
@@ -1,11 +1,16 @@
 # Supported and Tested VM Images
 
-* [HPC CentOS 7](#hpc-centos-7)
-* [HPC Rocky Linux 8](#hpc-rocky-linux-8)
-* [Debian 11](#debian-11)
-* [Ubuntu 20.04 LTS](#ubuntu-2004-lts)
-* [Windows](#windows)
+* [General Information](#general-information)
```
Comment (nick-stroud) [inline docs/vm-images.md:3]: This page was initially created to call out what images we officially support. I don't want to lose visibility of that. I propose the following title changes:

Supported and Tested VM Images -> VM Images  
General Information -> Specify an Image in a Blueprint  
VM Image Specifics -> HPC Toolkit Supported Images  

Indent `other images` and `slurm on gcp custom images` to be under `HPC Toolkit Supported Images`.

**Review by nick-stroud — CHANGES_REQUESTED** [review]

File: docs/vm-images.md [hunk 2]
```diff
@@ -14,7 +19,47 @@ Please see the [blueprint catalog](https://cloud.google.com/hpc-toolkit/docs/set
 
 For Slurm images, please see [SchedMD's GitHub repository](https://github.com/SchedMD/slurm-gcp/blob/master/docs/images.md#public-image).
 
-## HPC CentOS 7
+## General Information
+
+### Instance Images
+
+> **_Note:_** This information is applicable for most source modules, but there are some modules that have their own image specification.  Please read the documentation for any module utilized.
```
[the remark this answers is not in the document]
Comment (cdunbar13) [reply docs/vm-images.md]: Done

File: docs/vm-images.md [hunk 3]
```diff
@@ -14,7 +19,47 @@ Please see the [blueprint catalog](https://cloud.google.com/hpc-toolkit/docs/set
...
+
+When an HPC Toolkit blueprint points to a predefined source module (e.g. `community/modules/compute/schedmd-slurm-gcp-v5-node-group`), generally the module has a default image defined.  In order to override this default image, a user may specify the `instance_image` setting in the yaml blueprint, within either the specific module definition or the global variables.  The `instance_image` setting is defined by three parameters within the blueprint:
```
[the remark this answers is not in the document]
Comment (cdunbar13) [reply docs/vm-images.md]: Done

File: docs/vm-images.md [hunk 4]
```diff
@@ -14,7 +19,47 @@ Please see the [blueprint catalog](https://cloud.google.com/hpc-toolkit/docs/set
...
+
+```yaml
+instance_image:
+  project: centos-cloud
+  family: centos-v7        # If family is defined, omit name
+  name: centos-7-v20230809 # If name is defined, omit family
+```
+
+The `project` setting defines the space where the image will be found.  Either this is set to a known project where images are hosted (`centos-cloud`, `cloud-hpc-image-public`, `schedmd-slurm-public`, etc.) or to a team's or your own project where you have built your own images.
```
[the remark this answers is not in the document]
Comment (cdunbar13) [reply docs/vm-images.md]: Updated, let me know if you think the language is clear.

File: docs/vm-images.md [hunk 5]
```diff
@@ -14,7 +19,47 @@ Please see the [blueprint catalog](https://cloud.google.com/hpc-toolkit/docs/set
...
+
+The `family` setting defines a group of images built with the same label, and generally with some underlying similarities, usually an OS version or a software version installed on top of the OS.  When this is specified, instances will be created with the latest image within the family.  This will keep software more up to date, but will be less deterministic.
+
+The `name` setting defines a specific image.  These images are not likely to change frequently, but it cannot be guaranteed.
```
[the remark this answers is not in the document]
Comment (cdunbar13) [reply docs/vm-images.md]: I think I called it out more clearly.  If it needs to be specific examples (image publisher wants to fix a bug in an image or something), let me know.

File: docs/vm-images.md (hunk 1 above)
Comment (cdunbar13) [reply docs/vm-images.md]: Done.

**Review by nick-stroud — APPROVED** [review]

**Pull request merged**